lib/superset/services/duplicate_dashboard.rb

```diff
@@ -259,7 +259,7 @@ def source_dashboard_dataset_ids
       end
 
       def source_dashboard_filter_dataset_ids
-        @filter_dataset_ids ||= Superset::Dashboard::Filters::List.new(id).perform
+        @filter_dataset_ids ||= source_dashboard.filter_configuration.map { |c| c['targets'] }.flatten.compact.map { |c| c['datasetId'] }.flatten.compact
```

Comment:
@soundarya-mv 

Just wondering if we should do a uniq here?
incase the same dataset is used across multiple filters

Reply:
Done